marlowe/src/Language/Marlowe/Core/V1/Next/Applicables.hs

```diff
@@ -48,7 +49,13 @@ data ApplicableInputs
   deriving stock (Show,Eq,Ord,Generic)
   deriving anyclass (Pretty)
 
-
+filterByParties :: [Party] -> ApplicableInputs -> ApplicableInputs
+filterByParties parties ApplicableInputs {..}
+  = ApplicableInputs
+    { notifyMaybe = notifyMaybe
+    , deposits = filter (\(Indexed _ (CanDeposit party _ _ _ _)) -> elem party parties) deposits
+    , choices = filter (\(Indexed _ CanChoose {choiceId = (ChoiceId _ party)}) -> elem party parties) choices
```

Comment:
I think we want the default behaviour for `[]` parties to be no filtering. This will return nothing if no parties are provided.